airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py

```diff
@@ -137,21 +140,23 @@ def read_records(
             # parameter pointing to the future. In this case we dont need to
             # take any action and just return.
             return
+        profile = stream_slice["profile"]
         report_date = stream_slice[self.cursor_field]
-        report_infos = self._init_and_try_read_records(report_date)
+        report_infos = self._init_and_try_read_records(profile, report_date)
 
         for report_info in report_infos:
             for metric_object in report_info.metric_objects:
                 yield self._model(
                     profileId=report_info.profile_id,
                     recordType=report_info.record_type,
                     reportDate=report_date,
+                    updatedAt=pendulum.now(tz=profile.timezone).replace(microsecond=0).to_iso8601_string(),
                     metric=metric_object,
                 ).dict()
 
     def backoff_max_time(func):
         def wrapped(self, *args, **kwargs):
-            return backoff.on_exception(backoff.constant, RetryableException, max_time=self.report_wait_timeout)(func)(
+            return backoff.on_exception(backoff.constant, RetryableException, max_time=self.report_wait_timeout * 60, interval=10)(func)(
```

Comment:
Could you please give a little context of what this one is doing?

Reply:
it re-check statistic report generation status (IN_PROGRESS, COMPLETED)
if IN_PROGRESS it raise RetryableException excaption and re-check again

I just increased delay beatween retries from 1 second (default) to up to 10 seconds